x/gamm/keeper/grpc_query_test.go

```diff
@@ -283,3 +283,124 @@ func (suite *KeeperTestSuite) TestQueryBalancerPoolSpotPrice() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestEstimateSwapExactAmountIn() {
+	queryClient := suite.queryClient
+	poolID := suite.PrepareBalancerPool()
+
+	testCases := []struct {
+		name      string
+		req       *types.QuerySwapExactAmountInRequest
+		expectErr bool
+		result    sdk.Int
+	}{
+		{
+			name: "multi route",
```

Comment:
non-blocking comment / feel free to make an issue if you agree: only testing happy path here, we might want to test error cases as well

Reply:
added test cases for empty requests, I don't think error catching paths for estimate swap itself have to go here